specifyweb/frontend/js_src/lib/components/FormSliders/RecordSelectorFromIds.tsx

```diff
@@ -184,6 +184,35 @@ export function RecordSelectorFromIds<SCHEMA extends AnySchema>({
 
   const hasAttachments = tablesWithAttachments().includes(model);
 
+  const attachmentsObject = resource?.getDependentResource(
+    'collectionobjectattachments'
+  );
+  const attachmentModels = attachmentsObject?.models[0];
+  const dependentAttachment = attachmentModels?.dependentResources;
+  const attachmentObject = React.useMemo(
+    () =>
+      dependentAttachment === undefined ? [] : Object.keys(dependentAttachment),
+    [dependentAttachment]
+  );
+
+  const [attachmentIsPopulated, setAttachmentIsPopulated] =
+    React.useState(false);
+  const [trackChanges, setTrackChanges] = React.useState(0);
+  const hasChanged =
+    resource?.changed !== undefined &&
+    Object.keys(resource?.changed).length > 0 &&
+    attachmentIsPopulated;
+
+  React.useEffect(() => {
+    if (attachmentObject !== undefined && attachmentObject.length > 0) {
+      setAttachmentIsPopulated(true);
+    }
+
+    if (hasChanged) {
+      setTrackChanges((previousTrackChanges) => previousTrackChanges + 1);
+    }
+  }, [hasChanged, attachmentObject]);
```

Comment:
```suggestion
const attachmentsCollection = resource?.getDependentResource(
  'collectionobjectattachments'
);
const trackChanges = attachmentsCollection.length;
```

Reply:
this might or might not work. if it won't, then you have to add a useEffect that would add the 'change' listener on the attachmentsCollection and inside even handler update the value of trackChanges (trackChanges would be converted to a useState)

Reply:
unfortunately it doesn't work. I also tried with a useEffect but the changes to the attachments doesn't trigger the resource dependencies in the use effect, 
I also tried to create a ref with the length of attachmentsCollection and check if the ref is equal to the new ref and depending on that updating trackChanges but the issue is that the changes appear as soon as we click the + button to add an attachment instead of when the user save the attachment.

Reply:
yeah, two reasons the above didn't work:
- instead of hardcoding collectionobjectattachments, should use the <tableName>attachments
- since backbone resource changes don't trigger re-render, the code does not get re-executed when needed, thus you see stale state

solution would be:
- wait for migration away from backbone to make this all simple
- or add a useEffect that would be listening for collection to be created (because it may not exist right away, especially if you are opening the form for a new resource - http://localhost/specify/view/accession/new/), and then listen for changes to that collection, and on changes set the trackChanges state